app/src/main/java/org/mozilla/fenix/tabstray/inactivetabs/InactiveTabs.kt

```diff
@@ -0,0 +1,280 @@
+                            onIconClick = { onTabCloseClick(tab) },
+                        )
+                    }
+                }
+
+                Spacer(modifier = Modifier.height(8.dp))
+            }
+        }
+    }
+}
+
+/**
+ * Collapsible header for the Inactive Tabs section.
+ *
+ * @param expanded Whether the section is expanded.
+ * @param onClick Called when the user clicks on the header.
+ * @param onDeleteAllClick Called when the user clicks on the delete all button.
+ */
+@Composable
+private fun InactiveTabsHeader(
+    expanded: Boolean,
+    onClick: () -> Unit,
+    onDeleteAllClick: () -> Unit,
+) {
+    ExpandableListHeader(
+        headerText = stringResource(R.string.inactive_tabs_title),
+        expanded = expanded,
+        expandActionContentDescription = stringResource(R.string.inactive_tabs_expand),
+        collapseActionContentDescription = stringResource(R.string.inactive_tabs_collapse_group),
+        onClick = onClick,
+    ) {
+        IconButton(
+            onClick = onDeleteAllClick,
+            modifier = Modifier.padding(horizontal = 8.dp),
+        ) {
+            Icon(
+                painter = painterResource(R.drawable.ic_delete),
+                contentDescription = stringResource(R.string.inactive_tabs_delete_all),
+                modifier = Modifier.size(20.dp),
+                tint = FirefoxTheme.colors.iconPrimary,
+            )
+        }
+    }
+}
+
+/**
+ * Inactive Tabs auto close dialog.
+ *
+ * @param onDismissClick Called when the user clicks on the auto close dialog's dismiss button.
+ * @param onEnableAutoCloseClick Called when the user clicks on the auto close dialog's enable button.
+ */
+@Composable
+private fun InactiveTabsAutoClosePrompt(
+    onDismissClick: () -> Unit,
+    onEnableAutoCloseClick: () -> Unit,
+) {
+    Card(
+        modifier = Modifier.padding(horizontal = 16.dp, vertical = 12.dp),
+        shape = ROUNDED_CORNER_SHAPE,
+        backgroundColor = FirefoxTheme.colors.layer2,
+        border = BorderStroke(
+            width = 1.dp,
+            color = FirefoxTheme.colors.borderPrimary,
+        ),
+    ) {
+        Column(
+            modifier = Modifier.padding(start = 12.dp),
+            horizontalAlignment = Alignment.End,
+        ) {
+            Row(
+                verticalAlignment = Alignment.CenterVertically,
+            ) {
+                PrimaryText(
+                    text = stringResource(R.string.tab_tray_inactive_auto_close_title),
+                    modifier = Modifier.weight(1f),
+                    fontSize = 14.sp,
+                    fontFamily = FontFamily(Font(R.font.metropolis_semibold)),
+                )
+
+                IconButton(
+                    onClick = onDismissClick
+                ) {
+                    Icon(
+                        painter = painterResource(R.drawable.mozac_ic_close),
```

Comment:
We have a `mozac_ic_close_20` that we will use in the MessageCard and CollectionsPlaceholder https://github.com/mozilla-mobile/fenix/pull/24624. See https://www.figma.com/file/ujdiReypSypu2GhRgzArea/branch/SoVHeGS5UR99B7qFX1c8eK/Android-Components?node-id=2124%3A12124

Since you are using `Modifier.size(20.dp)` here, it might make sense to just replace it with this smaller close button and remove the modifier. However, I will note that the ic_close_20 is still very much a different icon in terms of look from ic_close. It does make sense to use ic_close_20 here because we already use it for all the other message looking cards that I mentioned above. I would double check with Crystal.

Reply:
In general, I don't ever expect to see a size modifier for an `Icon` since the internal implementation will use the size specified by the drawable. However, I think you did use the size here because of the IconButton's increase tap area, but you should specify the size on the IconButton instead and that won't cause an accessibility issue in my testing. 

To be doubly sure, I would test your entire inactive tabs with the accessibility scanner anyways.